### PR TITLE
Sync content immediately when content from parent project ready #8181

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncActivator.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSynchronizer.java  4
         

Complexity decreasing per file
==============================
+ modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncTask.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5822670&bid=19061950)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncActivator.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSynchronizer.java  2
         

Complexity decreasing per file
==============================
+ modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncTask.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5822670&bid=19061950)

File: modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    final Project sourceProject = this.projectService.list().
+                        stream().
+                        filter( project -> currentProjectName.equals( project.getName() ) ).
+                        findAny().
+                        orElse( null );
+
+                    this.projectService.list().
+                        stream().
+                        filter( project -> currentProjectName.equals( project.getParent() ) ).
+                        forEach( targetProject -> {
```
Comment (rymsha) [inline modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java:81]: Offload loops and heavy tasks into a separate thread (it still need to be in order, so for now single threaded thread pool would work)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncActivator.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSynchronizer.java  2
         

Complexity decreasing per file
==============================
+ modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncTask.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5822670&bid=19061950)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectNodeEventListener.java  3
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncActivator.java  1
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSynchronizer.java  2
         

Complexity decreasing per file
==============================
+ modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ParentProjectSyncTask.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5822670&bid=19061950)

**Pull request merged**